helpers/path.go

```diff
@@ -75,6 +75,35 @@ func (filepathBridge) Separator() string {
 
 var fpb filepathBridge
 
+// segmentReplacer replaces some URI-reserved characters in a path segments.
+var segmentReplacer = strings.NewReplacer("/", "-", "#", "-")
```

Comment:
Why do you replace "-" with "-"? (the same character).

Reply:
NewReplacer uses "old-new" pairs.  This replaces '/' and '#' with '-'.

Reply:
What if there is a `?` character? Is it possible to use a common url encoding here?

Reply:
See the tests below.  The `?` is removed.  URL encoding is beyond the scope of this PR.